### PR TITLE
fix: Sales Order Tracking ID Updates on Cancelation of DTI Shipment

### PR DESCRIPTION
**Pull request by bhattdevarsh**
[TASK](https://app.asana.com/0/1192407897953440/1199977102868348/f)

update_shipment_tracking_id

### CODE REVIEW

**Comment by bhattdevarsh** [comment]
@hrwX @sahil28297 @neilLasrado @vishdha, Please Review The PR

File: shipment_management/shipment_management/doctype/dti_shipment_note/dti_shipment_note.py [hunk 1]
```diff
@@ -57,7 +57,15 @@ def on_submit(self):
 		frappe.db.set(self, 'fedex_status', ShipmentNoteOperationalStatus.InProgress)
 		frappe.db.set(self, 'tracking_number', self.box_list[0].tracking_number)
 
-	# def on_cancel(self):
+	def on_cancel(self):
+		tracking_ids_list = frappe.get_value("Sales Order", self.sales_order, "tracking_ids").split(',')
```
Comment (hrwX) [inline shipment_management/shipment_management/doctype/dti_shipment_note/dti_shipment_note.py]:
```suggestion
		tracking_ids_list = frappe.db.get_value("Sales Order", self.sales_order, "tracking_ids").split(',')
```

Also you might have to strip every tracking id for better matching

File: shipment_management/shipment_management/doctype/dti_shipment_note/dti_shipment_note.py [hunk 2]
```diff
@@ -57,7 +57,15 @@ def on_submit(self):
...
+
+		for box in self.box_list:
+			if box.tracking_number in tracking_ids_list:
+				tracking_ids_list.remove(box.tracking_number)
```
Comment (hrwX) [inline shipment_management/shipment_management/doctype/dti_shipment_note/dti_shipment_note.py:65]: We need to remove the tracking_number not present in the box_list right ??

**Review by hrwX — CHANGES_REQUESTED** [review]

File: shipment_management/shipment_management/doctype/dti_shipment_note/dti_shipment_note.py (hunk 2 above)
Comment (bhattdevarsh) [reply shipment_management/shipment_management/doctype/dti_shipment_note/dti_shipment_note.py:65]: So when we cancel the shipment, the corresponding Tracking number(s) should be removed from the Sales Order Tracking IDs Field

**Review by hrwX — APPROVED** [review]

**Pull request merged**